Consistence/Sniffs/Exceptions/ExceptionDeclarationSniff.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Consistence\Sniffs\Exceptions;
+
+use PHP_CodeSniffer_File;
+use SlevomatCodingStandard\Helpers\ClassHelper;
+use SlevomatCodingStandard\Helpers\StringHelper;
+
+class ExceptionDeclarationSniff implements \PHP_CodeSniffer_Sniff
+{
+
+	const CODE_NOT_ENDING_WITH_EXCEPTION = 'NotEndingWithException';
+
+	/**
+	 * @return int[]
+	 */
+	public function register(): array
+	{
+		return [
+			T_CLASS,
+			T_INTERFACE,
+		];
+	}
+
+	/**
+	 * @param \PHP_CodeSniffer_File $phpcsFile
+	 * @param int $classPointer
+	 */
+	public function process(PHP_CodeSniffer_File $phpcsFile, $classPointer)
+	{
+		$extendedClassName = self::findExtendedClassName($phpcsFile, $classPointer);
+		if ($extendedClassName === false) {
+			return; //does not extend anything
+		}
+
+		if (!StringHelper::endsWith($extendedClassName, 'Exception')) {
+			return; // does not extend Exception, is not an exception
+		}
+
+		$className = ClassHelper::getName($phpcsFile, $classPointer);
+		if (!StringHelper::endsWith($className, 'Exception')) {
+			$phpcsFile->addError(sprintf(
+				'Exception class name "%s" must end with "Exception".',
+				$className
+			), $classPointer, self::CODE_NOT_ENDING_WITH_EXCEPTION);
+		}
+	}
+
+	/**
+	 * @see PHP_CodeSniffer_File::findExtendedClassName()
+	 * COPY & PASTE from there - but the interfaces are allowed here
+	 * Can be removed when https://github.com/squizlabs/PHP_CodeSniffer/pull/1473 is merged
```

Comment:
Did you mean separate class? Or separate method just for the interfaces?

I think we can leave the method here for now, but it should be private, not public.
Also please update the original doc (that it works with interfaces too) and move the `@see` section under it. And there are a few CS violations, could you fix them pls, because you never know, how long it will have to stay there :)